### PR TITLE
[BPK-3963] Upgrade Swift 5

### PR DESCRIPTION
**Pull request by imchristian**
+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] `UNRELEASED.md`
+ [ ] `README.md`
+ [ ] Tests
+ [ ] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/master/Example/Backpack%20Screenshot/Screenshots.swift)
+ [ ] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/master/Backpack/Backpack.h)
+ [ ] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Example/Podfile [hunk 1]
```diff
@@ -29,6 +29,15 @@ post_install do |installer|
     config.build_settings.delete('CODE_SIGNING_REQUIRED')
   end
 
+  installer.pods_project.targets.each do |target|
+    target.build_configurations.each do |config|
+        if target.name == "Backpack" || target.name == "FloatingPanel"
+            config.build_settings['SWIFT_VERSION'] = '5.0'
+        end
+        config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '12.0'
```
Comment (imchristian) [inline Example/Podfile:37]: This eliminates the current warnings 

`The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.0.99.`

File: Example/Podfile [hunk 2]
```diff
@@ -29,6 +29,15 @@ post_install do |installer|
     config.build_settings.delete('CODE_SIGNING_REQUIRED')
   end
 
+  installer.pods_project.targets.each do |target|
+    target.build_configurations.each do |config|
+        if target.name == "Backpack" || target.name == "FloatingPanel"
+            config.build_settings['SWIFT_VERSION'] = '5.0'
```
Comment (imchristian) [inline Example/Podfile:35]: Without this the pods reset to the previous swift versions every time we do `pod install` and it turns into a compile failure. [Some info about this](https://jeroenscode.com/migrating-to-swift-5/).

Comment (k0nserv) [reply Example/Podfile:35]: There's a way to indicate swift version via the `.podspec`. I feel like we should be using that and check if FloatingPanel hasn't added a way to do this too?

File: Example/Podfile (hunk 1 above)
Comment (k0nserv) [reply Example/Podfile:37]: I think the warnings are good to have? We can chase the projects that need to update this

**Review by k0nserv — COMMENTED** [review]
Can we have a CI job that runs on 4.2? This is what I have done for SkyFloatingLabelTextField, see https://github.com/Skyscanner/SkyFloatingLabelTextField/blob/master/.travis.yml

File: Backpack/BottomSheet/Classes/BackpackFloatingPanelController.swift [hunk 3]
```diff
@@ -69,7 +69,7 @@ private extension BPKFloatingPanelController {
         #if swift(>=4.2)
         addChild(bottomSection)
         #else
-        addChildViewController(bottomSection)
+        addChild(bottomSection)
```
Comment (georgegillams) [inline Backpack/BottomSheet/Classes/BackpackFloatingPanelController.swift:72]: Seems like this should already be ignored by Swift 5?
Maybe we can just remove this `if...else` if we are dropping support for `<= 4.2`?

File: Backpack.podspec [hunk 4]
```diff
@@ -43,5 +43,5 @@ Pod::Spec.new do |s|
   s.dependency 'MBProgressHUD', '~> 0.9.1'
   s.frameworks = 'UIKit', 'Foundation', 'CoreText'
   s.requires_arc = true
-  s.swift_version = '4.0'
+  s.swift_version = '5.0'
```
Comment (k0nserv) [inline Backpack.podspec:46]: We can and should provide an array here `swift_versions = ['5.0', '4.2', '4.0']`

File: .travis.yml [hunk 5]
```diff
@@ -15,9 +15,9 @@ cache:
 matrix:
   include:
     - osx_image: xcode12
-      env: BUILD_SDK=iphonesimulator14.0 DESTINATION="OS=14.0,platform=iOS Simulator,name=iPhone 8" FULL_TESTS="true" MAIN_STAGE="true"
+      env: SWIFT=5.0 BUILD_SDK=iphonesimulator14.0 DESTINATION="OS=14.0,platform=iOS Simulator,name=iPhone 8" FULL_TESTS="true" MAIN_STAGE="true"
     - osx_image: xcode12
-      env: BUILD_SDK=iphonesimulator14.0 DESTINATION="OS=13.5,platform=iOS Simulator,name=iPhone 8" FULL_TESTS="false" MAIN_STAGE="false"
+      env: SWIFT=4.2 BUILD_SDK=iphonesimulator14.0 DESTINATION="OS=13.5,platform=iOS Simulator,name=iPhone 8" FULL_TESTS="false" MAIN_STAGE="false"
```
Comment (k0nserv) [inline .travis.yml:20]: This isn't enough, if you look at https://github.com/Skyscanner/SkyFloatingLabelTextField/blob/master/Rakefile you'll see I am also passing down the swift verison

File: Backpack.podspec (hunk 4 above)
Comment (imchristian) [reply Backpack.podspec:46]: ohh I see, thanks for info 👍

File: .travis.yml (hunk 5 above)
Comment (imchristian) [reply .travis.yml:20]: I guess that's the missing part to pass CI then

File: Backpack.podspec (hunk 4 above)
Comment (k0nserv) [reply Backpack.podspec]: I think this should be **versions** with an `s` per the [docs](https://guides.cocoapods.org/syntax/podspec.html#swift_versions)

Comment (imchristian) [reply Backpack.podspec]: My change is `s.swift_versions` [here](https://github.com/Skyscanner/backpack-ios/pull/757/files#diff-ec850618833e8b72d1203858fa50fde5R46)

Comment (k0nserv) [reply Backpack.podspec]: Ah, I didn't notice that the diff was outdated. My bad

**Review by k0nserv — APPROVED** [review]

**Pull request merged**